webstruct/html_tokenizer.py

```diff
@@ -266,16 +311,23 @@ def _cleanup_elem(self, elem):
             elem.tail = self._tag_re.sub("", elem.tail)
 
     def _tokenize_and_split(self, text):
-        input_tokens = self._limit_tags(self.text_tokenize_func(text or ''))
-        input_tokens = map(six.text_type, input_tokens)
-        return self.sequence_encoder.encode_split(input_tokens)
+        text = text or ''
+        input_tokens = [t for t in self.text_tokenize_func(text)]
+        input_tokens = self._limit_tags(input_tokens)
+        input_tokens = [TextToken(chars=six.text_type(t.chars),
```

Comment:
unicode doesn't look right here; if t.chars if unicode (str in Python 3) then conversion is not needed; if t.chars is bytes, then conversion should use a proper encoding, not sys.getdefaultencoding() which is often ascii

Reply:
This conversion is the same as it was [before](https://github.com/scrapinghub/webstruct/blob/c2af27808c27a5ffc8cdd40c921408aa9d3dcfc6/webstruct/html_tokenizer.py#L270)

Reply:
One of [tests](https://github.com/scrapinghub/webstruct/blob/c2af27808c27a5ffc8cdd40c921408aa9d3dcfc6/webstruct/tests/test_html_tokenizer.py#L59) awaits unicode

Reply:
All real encoding/decoding handled by lxml. lxml uses utf-8 as it is internal representation. I think we can add test with real unicode and safely remove this conversion

Reply:
Ah, I finally recalled why do we have such code! In Python 2.x lxml returns bytes for ASCII-only data and unicode for non-ascii data; this code ensures everything is unicode. It is only active for ascii-only bytes in Python 2.x, and no-op in all other cases, so it works as intended. Sorry for a false alarm.